man/spek.1
.\" Automatically generated by Pandoc 2.9.2.1
.\"
.TH "SPEK" "1" "2023-01-09" "User\[cq]s Guide" "Version 0.8.5"
.hy
.SH NAME
.PP
Spek - Acoustic Spectrum Analyser
.SH SYNOPSIS
.PP
\f[C]spek\f[R] [\f[I]OPTION\f[R] \f[I]\&...\f[R]] [\f[I]FILE\f[R]]
.SH DESCRIPTION
.PP
\f[I]Spek\f[R] generates a spectrogram for the input audio file.
.SH OPTIONS
.TP
\f[B]\f[CB]-h\f[B]\f[R], \f[B]\f[CB]--help\f[B]\f[R]
Output the help message then quit.
.TP
\f[B]\f[CB]-V\f[B]\f[R], \f[B]\f[CB]--version\f[B]\f[R]
Output version information then quit.
.SH KEYBINDINGS
.SS Notes
.PP
On OS X use the Command key instead of Ctrl.
.SS Menu
.TP
\f[B]\f[CB]Ctrl-O\f[B]\f[R]
Open a new file.
.TP
\f[B]\f[CB]Ctrl-S\f[B]\f[R]
Save the spectrogram as an image file.
.TP
\f[B]\f[CB]Ctrl-E\f[B]\f[R]
Show the preferences dialog.
.TP
\f[B]\f[CB]F1\f[B]\f[R]
Open online manual in the browser.
.TP
\f[B]\f[CB]Shift-F1\f[B]\f[R]
Show the about dialog.
.SS Spectrogram
.TP
\f[B]\f[CB]c\f[B]\f[R], \f[B]\f[CB]C\f[B]\f[R]
Change the audio channel.
.TP
\f[B]\f[CB]f\f[B]\f[R], \f[B]\f[CB]F\f[B]\f[R]
Change the DFT window function.
.TP
\f[B]\f[CB]l\f[B]\f[R], \f[B]\f[CB]L\f[B]\f[R]
Change the lower limit of the dynamic range in dBFS.
.TP
\f[B]\f[CB]p\f[B]\f[R], \f[B]\f[CB]P\f[B]\f[R]
Change the palette.
.TP
\f[B]\f[CB]s\f[B]\f[R], \f[B]\f[CB]S\f[B]\f[R]
Change the audio stream.
.TP
\f[B]\f[CB]u\f[B]\f[R], \f[B]\f[CB]U\f[B]\f[R]
Change the upper limit of the dynamic range in dBFS.
.TP
\f[B]\f[CB]w\f[B]\f[R], \f[B]\f[CB]W\f[B]\f[R]
Change the DFT window size.
.SH FILES
.TP
\f[I]\[ti]/.config/spek/preferences\f[R]
The configuration file for \f[I]Spek\f[R], stored in a simple INI
format.
.SH AUTHORS
.PP
Alexander Kojevnikov <alexander@kojevnikov.com>.
Other contributors are listed in the LICENCE.md file distributed with
the source code.
.SH DISTRIBUTION
.PP
The latest version of \f[I]Spek\f[R] may be downloaded from
<https://www.spek.cc>.


NEWS
Spek 0.7 - Released 2011-04-24
==============================================================================

New Features And Enhancements
------------------------------------------------------------------------------

Spek 0.7 is part of beta 0.x releases, leading up to a stable 1.0 release
later this year. Check Spek website for the roadmap.

New features since 0.6:

 * Added translations in 8 languages
 * Preferences to select a language and to check for a new version
 * Spectral density ruler
 * Added a menu bar, cleaned up the tool bar.
 * Better OS X integration

Enhancements:

 * Show the version number in the window
 * Pre-compute the cosine table to speed up analysis by ±16%
 * Use jhbuild and ige-mac-bundler to build and package Spek on OS X
 * spek(1) man page
 * Avoid using APIs depreciated in GTK3

Bugfixes:

 * Fixed link activation on OS X (issue 31)
 * Fixed new version detection on OS X
 * Fixed duration for unsynchronised ID3v24 mp3 tags (upstream fix)
 * Fixed rigth click → Quit (issue 24) and the ⌘ Q shortcut (issue 44)

Sources / Packages
------------------------------------------------------------------------------

 * Spek 0.7 tarball:
   * <http://spek.googlecode.com/files/spek-0.7.tar.bz2>

 * Windows and Mac OS X binaries:
   * <http://spek.googlecode.com/files/spek-0.7.msi>
   * <http://spek.googlecode.com/files/spek-0.7.zip>
   * <http://spek.googlecode.com/files/spek-0.7.dmg>

 * Ubuntu PPA
   * <https://launchpad.net/~alexk/+archive/spek>

 * Other Unix packages:
   * <http://code.google.com/p/spek/wiki/UnixInstall>

Dependencies
------------------------------------------------------------------------------

 * GTK+ >= 2.18
 * FFmpeg libraries:
   * libavcodec >= 52.56.0 (r22291)
   * libavformat
   * libavutil


src/spek.cc
#include <wx/cmdline.h>
#include <wx/log.h>
#include <wx/socket.h>

#include "spek-artwork.h"
#include "spek-platform.h"
#include "spek-preferences.h"

#include "spek-window.h"

class Spek: public wxApp
{
public:
    Spek() : wxApp(), window(NULL), quit(false) {}

protected:
    virtual bool OnInit();
    virtual int OnRun();
#ifdef OS_OSX
    virtual void MacOpenFiles(const wxArrayString& files);
#endif

private:
    SpekWindow *window;
    wxString path;
    wxString device;
    bool quit;
};

IMPLEMENT_APP(Spek)

bool Spek::OnInit()
{
    delete wxLog::SetActiveTarget(new wxLogStderr()) ;
    wxInitAllImageHandlers();
    wxSocketBase::Initialize();

    spek_artwork_init();
    spek_platform_init();
    SpekPreferences::get().init();

    static const wxCmdLineEntryDesc desc[] = {{
            wxCMD_LINE_SWITCH,
            "h",
            "help",
            "Show this help message",
            wxCMD_LINE_VAL_NONE,
            wxCMD_LINE_OPTION_HELP,
        }, {
            wxCMD_LINE_SWITCH,
            "V",
            "version",
            "Display the version and exit",
            wxCMD_LINE_VAL_NONE,
            wxCMD_LINE_PARAM_OPTIONAL,
        }, {
            wxCMD_LINE_SWITCH,
            NULL,
            "microphone",
            "Show continuous graph recorded from mircophone",
            wxCMD_LINE_VAL_NONE,
            wxCMD_LINE_PARAM_OPTIONAL,
        }, {
            wxCMD_LINE_PARAM,
            NULL,
            NULL,
            "FILE",
            wxCMD_LINE_VAL_STRING,
            wxCMD_LINE_PARAM_OPTIONAL,
        },
        wxCMD_LINE_DESC_END,
    };

    wxCmdLineParser parser(desc, argc, argv);
    int ret = parser.Parse(true);
    if (ret == 1) {
        return false;
    }
    if (ret == -1) {
        this->quit = true;
        return true;
    }
    if (parser.Found("version")) {
        // TRANSLATORS: the %s is the package version.
        wxPrintf(_("Spek version %s"), PACKAGE_VERSION);
        wxPrintf("\n");
        this->quit = true;
        return true;
    }
    if (parser.Found("microphone")) {
        this->device = "default";
    }
    if (parser.GetParamCount()) {
        this->path = parser.GetParam();
    }

    this->window = new SpekWindow(this->path, this->device);
    this->window->Show(true);
    SetTopWindow(this->window);
    return true;
}

int Spek::OnRun()
{
    if (quit) {
        return 0;
    }

    return wxApp::OnRun();
}

#ifdef OS_OSX
void Spek::MacOpenFiles(const wxArrayString& files)
{
    if (files.GetCount() == 1) {
        this->window->open(files[0]);
    }
}
#endif
